offices/tests/test_models.py

```diff
@@ -0,0 +1,69 @@
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+from django.db import IntegrityError
+from django.test import TestCase
+
+from offices.models import Office, Company
+
+
+class CompanyModelTest(TestCase):
+
+    def set_up(self):
+        self.first_office = Office.objects.create(country="test",
+                                                  street="test",
+                                                  postal_code="123qwe",
+                                                  city="test",
+                                                  monthly_rent=12)
+        self.second_office = Office.objects.create(country="test",
+                                                   street="test",
+                                                   postal_code="123qwe",
+                                                   city="test",
+                                                   monthly_rent=12)
+
+    def test_company_should_always_have_a_headquarter(self):
+        self.assertRaises(ValidationError,
+                          lambda: Company.objects.create(name="test"))
+
+    def test_company_should_always_have_exact_one_headquarter(self):
+        self.set_up()
+        test_company = Company.objects.create(name='test1',
+                                              headquarter=self.first_office)
+
+        def set_another_headquarter():
+            test_company.headquarter = self.second_office
+            test_company.save()
+            return Office.objects.get(
+                pk=self.first_office.id).has_is_headquarter()
+
+        self.assertEqual(False, set_another_headquarter())
+
+    def test_company_cant_exist_without_headquarter(self):
```

Comment:
I'm not sure what you test at here - you said `company can not exist without headquater`, but you are checking Office instead of Company

Reply:
Yes, It's a little strange logic. But on that moment if we delete company it's headquarter should be deleted too. So it should not exist.